packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts

```diff
@@ -61,25 +62,28 @@ export const saveEnvironmentVariables = (resourceName: string, newEnvironmentVar
  */
 export const askEnvironmentVariableCarryOut = async (
   context: $TSContext,
-  currentEnvName: string,
-  projectPath: string,
+  fromEnvName: string,
   yesFlagSet?: boolean,
 ): Promise<void> => {
-  const teamProviderInfo = stateManager.getTeamProviderInfo(projectPath, {
-    throwIfNotExist: false,
-    default: {},
-  });
-  const functions = teamProviderInfo[currentEnvName]?.categories?.function || {};
-  const functionNames = Object.keys(functions);
+  await ensureEnvParamManager(fromEnvName);
+  await ensureEnvParamManager();
```

Comment:
why this is called 2 times ?

Reply:
This code is copying parameters from one environment to another so the param manager for both environments needs to be initialized. The first call initializes it for the "from" environment and the second call initializes it for the current ("to") environment